src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,279 @@
+        int timeFullSeparation;
+
+        if ((timeFullSeparation = timeFull.indexOf(" ")) != -1) {
+            String time = timeFull.substring(timeFullSeparation + 1);
+            LocalDate date = LocalDate.parse(timeFull.substring(0, timeFullSeparation));
+            taskList.add(new Deadline(taskName, date, time));
+        } else {
+            LocalDate date = LocalDate.parse(timeFull);
+            taskList.add(new Deadline(taskName, date));
+        }
+
+        printTaskAdded(taskName);
+    }
+
+    /**
+     * Adds the event entered by the user to the list and prints it.
+     *
+     * @param input The event inputted by the user.
+     * @throws MissingTaskException If task is unspecified after command.
+     * @throws MissingTimeException If time is unspecified after command.
+     * @throws DateTimeParseException If date entered is of the wrong format.
+     */
+    public static void addEvent(String input)
+            throws MissingTaskException, MissingTimeException, DateTimeParseException {
+        int separation = input.indexOf(" /at ");
+
+        if (separation == -1) {
+            throw new MissingTimeException("Time not found");
+        }
+
+        if (separation < 8) {
+            throw new MissingTaskException("Task not found");
+        }
+        String taskName = input.substring(6, separation);
+
+        if (input.substring(separation + 6).length() < 1) {
+            throw new MissingTimeException("Time not found");
+        }
+
+        String timeFull = input.substring(separation + 5);
+        int timeFullSeparation;
+
+        if ((timeFullSeparation = timeFull.indexOf(" ")) != -1) {
+            String time = timeFull.substring(timeFullSeparation + 1);
+            LocalDate date = LocalDate.parse(timeFull.substring(0, timeFullSeparation));
+            taskList.add(new Event(taskName, date, time));
+        } else {
+            LocalDate date = LocalDate.parse(timeFull);
+            taskList.add(new Event(taskName, date));
+        }
+
+        printTaskAdded(taskName);
+    }
+
+    /**
+     * Prints the confirmation of the addition of the last task.
+     *
+     * @param taskName The name of the task just added.
+     */
+    public static void printTaskAdded(String taskName) {
+        System.out.printf("Gotcha! The following task has been added:\n"
+                + "\t%s\n"
+                + "You now have %d "
+                + (taskList.size() == 1 ? "task" : "tasks")
+                + " in your list!\n",
+                taskName, taskList.size());
+    }
+
+    /**
+     * Finds the tasks requested by the user and prints them.
+     *
+     * @param input The event inputted by the user.
+     * @throws MissingTaskException If task is unspecified after command.
+     * @throws MissingTimeException If time is unspecified after command.
+     */
+    public static void findTask(String input) throws MissingTaskException {
+        ArrayList<Task> toPrint = new ArrayList<>();
```

Comment:
Should the `toPrint` variable have the word 'task' in its name? I feel like it would be clearer that the variable is referring to a `TaskList` that way.